### PR TITLE
Mattermost support

### PR DESCRIPTION
**Pull request by dbrgn**
It should even be compatible with Slack, although [outgoing webhooks are deprecated](https://api.slack.com/custom-integrations/outgoing-webhooks).

TODO:

- [x] Move all common functionality into standalone functions
- [x] Integrate that stuff into Mattermost
- [x] Add unit tests
- [x] Test against Mattermost

### CODE REVIEW

**Comment by dbrgn** [comment]
screenshot

Ready for review!

File: .circleci/config.yml [hunk 1]
```diff
@@ -37,11 +38,53 @@ jobs:
...
+          requires:
+            - test-node8
+            - test-node10
+            - lint
+          filters:
+            branches:
+              only:
+                - master
+
+  # Build every week on Monday at 04:00 am
```
Comment (rnestler) [inline .circleci/config.yml:83]: What's the point in that? Getting updates from the base image?

Comment (dbrgn) [reply .circleci/config.yml:83]: Exactly :)

File: src/bot.ts [hunk 2]
```diff
@@ -1,14 +1,42 @@
...
+  process.exit(1);
+}
+if (telegramToken.length === 0) {
+  console.error('Please set TELEGRAM_TOKEN env var');
+  process.exit(1);
+}
+if (mattermostToken.length === 0) {
+  console.error('Please set MATTERMOST_TOKEN env var');
+  process.exit(1);
+}
```
Comment (rnestler) [inline src/bot.ts:31]: Shouldn't you verify the other config values (port, and debug) as well?

**Review by rnestler — APPROVED** [review]
Just one small comment, otherwise LGTM.

File: src/bot.ts (hunk 2 above)
Comment (dbrgn) [reply src/bot.ts:31]: No, they both have default values if not set.

Full validation isn't necessary either, the program will simply not start and show an error message (e.g. if the type is wrong) :slightly_smiling_face:

**Pull request merged**